UnityProject/Assets/Scripts/UI/Systems/ServerInfoPanel/Pages/ChangelogPage.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System.Collections;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using UI.Systems.ServerInfoPanel.Models;
+using UnityEngine;
+using UnityEngine.UI;
+using Util.Rx;
+
+namespace UI.Systems.ServerInfoPanel
+{
+	public class ChangelogPage: InfoPanelPage
+	{
+		[SerializeField] private Button previousButton;
+		[SerializeField] private Button nextButton;
+		[SerializeField] private GameObject buildEntryPrefab;
+		[SerializeField] private Transform changesContainer;
+
+		private const string BASE_API_URL = "https://changelog.unitystation.org/all-changes";
+
+		private readonly BehaviourSubject<AllChangesResponse> changelogData = new(null);
+		private string currentApiUrl = BASE_API_URL;
+
+
+		public void RefreshPage()
+		{
+			DestroyAllEntries();
+			RefreshPaginationButtons();
+			if (changelogData.Value == null) return;
+
+
+			StartCoroutine(SpawnPrefabs());
+		}
+
+		private IEnumerator SpawnPrefabs()
+		{
+			if (changelogData.Value == null) yield break;
+
+			//Separate the entries into chunks of 10
+			foreach (var chunk in changelogData.Value.results.Chunk(10))
+			{
+				foreach (var build in chunk)
+				{
+					var buildEntry = Instantiate(buildEntryPrefab, changesContainer);
+					buildEntry.GetComponent<BuildEntry>().SetBuild(build);
+				}
+
+				//wait for next frame to keep spawning
+				yield return null;
+			}
+		}
+
+		private void DestroyAllEntries()
+		{
+			foreach (Transform child in changesContainer)
+			{
+				Destroy(child.gameObject);
+			}
+		}
+
+
+		private void RefreshPaginationButtons()
+		{
+			previousButton.interactable = changelogData?.Value?.previous != null;
+			nextButton.interactable = changelogData?.Value?.next != null;
+		}
+
+		public override bool HasContent()
+		{
+			return true;
+		}
+
+		private void ChangeDataPage(string newPage)
+		{
+			if (string.IsNullOrEmpty(newPage)) return;
+			currentApiUrl = newPage;
+			FetchChanges(newPage).Then(
+				(newData) => { changelogData.Next(newData.Result); }
+			);
+		}
+
+		public void OnNextButtonClicked()
+		{
+			ChangeDataPage(changelogData.Value.next);
+		}
+
+		public void OnPreviousButtonClicked()
+		{
+			ChangeDataPage(changelogData.Value.previous);
+		}
+
+		private static async Task<AllChangesResponse> FetchChanges(string url = BASE_API_URL)
+		{
+			AllChangesResponse newData = null;
+			using var client = new HttpClient();
+			client.DefaultRequestHeaders.Accept.Add(
+				new MediaTypeWithQualityHeaderValue("application/json"));
+			using var response = client.GetAsync(url).Result;
```

Comment:
This still blocks on the web request.

```suggestion
			using var response = await client.GetAsync(url);
```

Reply:
Aight, gonna fix it